lib/philomena/images.ex

```diff
@@ -427,6 +431,44 @@ defmodule Philomena.Images do
         {:ok, count}
     end)
     |> Repo.transaction()
+    |> case do
+      {:ok, %{image: {image, _added, _removed}}} = res ->
+        update_tag_change_limits_after_commit(image, attribution)
+
+        res
+
+      err ->
+        err
+    end
+  end
+
+  defp check_tag_change_limits_before_commit(image, attribution) do
+    user = attribution[:user]
+    ip = attribution[:ip]
+
+    cond do
+      Limits.is_limited_for_tag_count?(user, ip) ->
+        {:error, "Limit exceeded"}
+
+      Limits.is_limited_for_rating_count?(user, ip) ->
```

Comment:
This needs to check if the current batch contains rating tag changes, otherwise it'll fail the update if user changed rating tag somewhere, but not in this batch.